crates/iota/src/client_commands.rs

```diff
@@ -1415,6 +1415,11 @@ impl IotaClientCommands {
                     None => context.config().keystore().get_alias_by_address(&address)?,
                 };
 
+                if context.config().active_address().is_none() {
+                    context.config_mut().set_active_address(address);
+                    context.config().save()?;
+                }
```

Comment:
Was this the only missing bit? Are the changes to `new` and `set_envs` actually required?

Reply:
Not required, they just do the same for envs basically.

Reply:
Why not do the same for envs here then?

Reply:
because this isn't where the env is set

Reply:
Ah, we're in NewAddress, my bad